### PR TITLE
feat(UI): Renew password when expired

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds a new page to change a password expired.
This also displays a warning message, in the user top counter, when the password will expire in less than 7 days.

Screenshot 2022-02-15 at 15 21 04

Screenshot 2022-02-25 at 15 58 11

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Pre-requisite: Your password is expired

- In the login page, enter your credentials
- Click on the "Connect" button
- -> You are redirected to the reset password page
- Type your current password
- Type the same password as the old one in the new password field
- -> An error appear
- Type a new password
- Confirm your with a mispell
- -> Another error appear
- Confirm your new password again
- Click on the "Reset password" button
- -> You are redirect to your default page
- When you"re password is about to expire (< 7 days), it displays a message in the user top counter

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Login/index.test.tsx [hunk 1]
```diff
@@ -218,4 +239,20 @@ describe('Login Page', () => {
     );
     expect(screen.getByLabelText(labelHideThePassword)).toBeInTheDocument();
   });
+
+  it('redirects to the user to the reset page when the submitted password is expired', async () => {
```
Comment (bdauria) [inline www/front_src/src/Login/index.test.tsx:243]:
```suggestion
  it('redirects to the reset page when the submitted password is expired', async () => {
```

**Review by bdauria — DISMISSED** [review]

**Review by bdauria — DISMISSED** [review]

File: src/Core/Application/Security/Exception/UserPasswordException.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace Core\Application\Security\Exception;
+
+class UserPasswordException extends \Exception
+{
+    /**
+     * Exception thrown when a password doesn't match the security policy.
+     *
+     * @return self
+     */
+    public static function passwordDoesnotMatchSecurityPolicy(): self
```
Comment (callapa) [inline src/Core/Application/Security/Exception/UserPasswordException.php:33]:
```suggestion
    public static function passwordDoesNotMatchSecurityPolicy(): self
```

**Review by a-launois — APPROVED** [review]
BE ok

**Review by callapa — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]
OK FE part

**Pull request merged**